frames/src/main/java/com/checkout/frames/screen/billingaddress/billingaddressdetails/BillingAddressDetailsViewModel.kt

```diff
@@ -56,40 +68,96 @@ internal class BillingAddressDetailsViewModel @Inject constructor(
     val inputComponentsContainerModifier =
         containerMapper.map(style.inputComponentsContainerStyle.containerStyle)
 
-    val inputComponentsState = provideBillingAddressInputComponentsState(
-        style
-    )
-    val inputComponentsViewContainerStyle = provideBillingAddressInputComponentsViewContainerStyle(
-        style
-    )
+    val inputComponentsStateList = provideInputComponentStateList(style)
 
-    private fun provideBillingAddressInputComponentsViewContainerStyle(
-        billingAddressDetailsStyle: BillingAddressDetailsStyle
-    ): BillingAddressInputComponentsViewContainerStyle {
-        return billingAddressDetailsComponentStyleUseCase.execute(billingAddressDetailsStyle)
+    val inputComponentViewStyleList = provideInputComponentViewStyleList(style)
+
+    private var wasFocused = false
+
+    internal companion object {
+        val onlyDigitsRegex = "[^0-9]".toRegex()
     }
 
-    private fun provideBillingAddressInputComponentsState(billingAddressDetailsStyle: BillingAddressDetailsStyle):
-            BillingAddressInputComponentsContainerState {
-        return billingAddressDetailsComponentStateUseCase.execute(
-            billingAddressDetailsStyle
-        )
+    init {
+        viewModelScope.launch {
+            isReadyToSaveAddress().collect {
+                screenButtonState.isEnabled.value = it
+            }
+        }
+
+        inputComponentsStateList.forEachIndexed { index, billingAddressInputComponentState ->
+            with(inputComponentsStateList[index]) {
+                addressFieldText.value = paymentStateManager.billingAddress.value
+                    .updateAddressFieldText(billingAddressInputComponentState.addressFieldName)
+                if (addressFieldText.value.isNotBlank() && !isInputFieldOptional) {
+                    isAddressFieldValid.value = true
+                }
+            }
+        }
```

Comment:
This block of code is not really self-describable.
Please either put it to a separate function ie. `private fun updateInitialState` or write a comment.